### PR TITLE
Add Elementary and Polycyclic groups in Permutation groups

### PR DESCRIPTION
**Pull request by divyanshu132**
#### Brief description of what is fixed or changed
`Elementary` and `Polycyclic` groups has been implemented where `Elementary Group` is defined as a finite `abelian group`, where every `nontrivial` element has an order of `p`, where `p` is a `prime` and `Polycyclic Group` for a finite group is same as `solvable group`. One TODO task may be implementing polycyclic_group for infiite groups.

#### Release Notes

*  combinatorics
   *  elementary and polycyclic groups has been added to perm_groups.py

### CODE REVIEW

**Comment by divyanshu132** [comment]
@jksuom please review.

**Comment by jksuom** [comment]
It would be useful to have the prime `p` in the API. So, for instance, one could check if a Sylow p-group is elementary. The signature could be `is_elementary(self, p)` ('group' can be omitted).

All permutation groups implemented in SymPy are finite. There only remains to check that the group is abelian and that the generators have order `p`. So something like this should work:

    return self.is_abelian and all(g.order() == p for g in self.generators)

As finite groups are polycyclic if and only if they are solvable, these properties can be defined by the same code:

    is_polycyclic = is_solvable

**Comment by divyanshu132** [comment]
Will it be good that user have to specify `p`?

**Comment by divyanshu132** [comment]
@jksuom I've updated as you suggested, please have a look.

File: sympy/combinatorics/perm_groups.py [hunk 1]
```diff
@@ -2073,6 +2098,24 @@ def is_subgroup(self, G, strict=True):
             return False
         return all(G.contains(g, strict=strict) for g in gens)
 
+    def is_polycyclic(self):
```
Comment (jksuom) [inline sympy/combinatorics/perm_groups.py]: Perhaps this could be a property.

Comment (divyanshu132) [reply sympy/combinatorics/perm_groups.py]: Done!!

**Comment by jksuom** [comment]
> Will it be good that user have to specify p?

When working with p-groups, for instance Sylow groups, the prime p is usually known, so it should not be hard to specify. Here are some typical examples of groups of order a power of p = 2.
```
In [2]: G = SymmetricGroup(4).sylow_subgroup(2)

In [3]: G.order()
Out[3]: 8

In [4]: G.is_elementary(2)
Out[4]: False

In [5]: H = AlternatingGroup(4).sylow_subgroup(2)

In [6]: H.order()
Out[6]: 4

In [7]: H.is_elementary(2)
Out[7]: True
```
(These could be added to tests or doctests.)

**Comment by divyanshu132** [comment]
Added above tests.

**Comment by divyanshu132** [comment]
@jksuom can this be merged.

File: sympy/combinatorics/perm_groups.py [hunk 2]
```diff
@@ -155,6 +155,7 @@ def __new__(cls, *args, **kwargs):
         obj._is_trivial = None
         obj._transitivity_degree = None
         obj._max_div = None
+        obj._is_polycyclic = None
```
Comment (jksuom) [inline sympy/combinatorics/perm_groups.py]: I think this can be omitted. It is just a copy of `_is_solvable`.

Comment (divyanshu132) [reply sympy/combinatorics/perm_groups.py]: okay

**Comment by divyanshu132** [comment]
@jksuom I think there is something wrong with the master branch because when I merged master in this branch, the travis build fails please have a look.

**Comment by jksuom** [comment]
There are errors like this:
> CondaHTTPError: HTTP 500 INTERNAL SERVER ERROR for url <https://conda.anaconda.org/conda-forge/label/cf201901/linux-64/ipython-5.8.0-py36_0.tar.bz2>

I don't think there is anything we could do to fix those http 500 server errors, except wait.

**Comment by jksuom** [comment]
Please don't close and reopen. It will restart all jobs, even those that have passed already. I'll restart the failing job when the server has been fixed.

**Comment by divyanshu132** [comment]
Okay, thanks.

**Comment by jksuom** [comment]
Thanks, the error was fixed and this can be merged.

**Comment by divyanshu132** [comment]
Thanks, @jksuom Does it make sense to add [Mathieu Group](https://en.wikipedia.org/wiki/Mathieu_group) in `SymPy` because it works only for groups with degree `9, 10, 11, 12, 21, 22, 23, 24`.

**Pull request merged**